### PR TITLE
Fix multiple sensor data load bug .

### PR DESCRIPTION
**Pull request by hardik124**
Modified adapter declaration for better performance.
Shifted Sensor data loading code in a method to avoid code replication,

Implemented requested changes.

Made "data" a local variable following up codacy's recommendation

Squashed all commits

Fixes issue 
#457

Apk [here](https://ufile.io/jrl96)

### CODE REVIEW

File: app/src/main/java/org/fossasia/pslab/sensorfragment/SensorFragmentMain.java [hunk 1]
```diff
@@ -55,37 +53,18 @@ public void onCreate(Bundle savedInstanceState) {
...
-                    }
-                }
-                for (String s : dataAddress) {
-                    tvData += s + ":" + sensorAddr.get(Integer.parseInt(s)) + "\n";
-                }
-            }
-            String[] dataDisp = dataName.toArray(new String[dataName.size()]);
-            adapter = new ArrayAdapter<>(getContext(), R.layout.sensor_list_item, R.id.tv_sensor_list_item, dataDisp);
-        }
+        i2c = scienceLab.i2c;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/sensorfragment/SensorFragmentMain.java]: Hello @hardik124 ! Seems like a very nice fix! Thanks for the PR! :) Wouldn't this line throw any error if the device is not connected?

Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/sensorfragment/SensorFragmentMain.java]: Nope its only referencing to ```i2c``` variable of sciencelab which is ```public```.

**Comment by viveksb007** [comment]
@hardik124 from next time please upload apk directly on github by creating zip or upload on drive and provide a link.

**Comment by hardik124** [comment]
Yeah . Sure.
Actually it would have taken another commit so I thought of uploading
somewhere else😅

**Comment by viveksb007** [comment]
No you don't have to add your apk to repo.
Just zip it and drag it over to comment section, it would upload.

**Comment by hardik124** [comment]
I will keep that in mind.

**Comment by viveksb007** [comment]
@hardik124 Please provide updated apk for testing. The one attached above is before you made changes in your PR.

**Comment by hardik124** [comment]
[Uploading app-debug.apk.zip…](Apk here.)

**Comment by hardik124** [comment]
Apk here :)
[app-debug.apk.zip](https://github.com/fossasia/pslab-android/files/1211354/app-debug.apk.zip)

**Comment by viveksb007** [comment]
Works great @hardik124

**Comment by hardik124** [comment]
Thanks.

**Pull request merged**